### PR TITLE
Feature/refactor + dependabot

### PR DESCRIPTION
**Pull request by CarlosGomes98**
- Refactor directory structure to remove the unnecessary `src` directory
- Set up dependabot 
- Added missing copyright headers (that is the bulk of the changes)

### CODE REVIEW

**Comment by CarlosGomes98** [comment]
After merging, dependabot version updates should be enabled in the repo settings

File: terratorch/datamodules/__init__.py
Comment (Joao-L-S-Almeida) [inline terratorch/datamodules/__init__.py:1]: These datamodules are being imported but not used, right ?

File: terratorch/datasets/__init__.py
Comment (Joao-L-S-Almeida) [inline terratorch/datasets/__init__.py:1]: Same question asked for the other __init__ file.

File: .github/workflows/test.yaml
Comment (Joao-L-S-Almeida) [inline .github/workflows/test.yaml:1]: In which circumstances we want to perform the tests for individual pushes ? As we have the rule of creating PR before sending pushes to `main` I thought it was already covered by the pull request instruction.

**Comment by Joao-L-S-Almeida** [comment]
In general, how the massive renaming from src/terratorch to terratorch will affect the still open PR's  (specially https://github.com/IBM/terratorch/pull/19) ?

**Comment by CarlosGomes98** [comment]
> In general, how the massive renaming from src/terratorch to terratorch will affect the still open PR's (specially #19) ?

@Joao-L-S-Almeida This is a good point. Should we hold off on this one until the currently open PRs are merged, and then do it? This should help minimize any headaches from it

File: .github/workflows/test.yaml
Comment (CarlosGomes98) [reply .github/workflows/test.yaml:1]: Its a good point. Tbh I mostly took this from what I saw on TorchGeo. Could it be for the case where a force push to main happens?

File: terratorch/datamodules/__init__.py
Comment (CarlosGomes98) [reply terratorch/datamodules/__init__.py:1]: They are not being used, but they enable `from terratorch.datamodules import`, rather than needing to go down to the module

File: terratorch/datasets/__init__.py
Comment (CarlosGomes98) [reply terratorch/datasets/__init__.py:1]: Yes, same thing.

File: .github/workflows/test.yaml
Comment (Joao-L-S-Almeida) [reply .github/workflows/test.yaml:1]: Right. I believe it won't affect our current workflow, but it could cover the exceptional cases in which we need to direclty push to the master. I agree.

**Comment by Joao-L-S-Almeida** [comment]
I believe just #19 may be affected by these changes. Maybe we could consider to approve or discard it before approving with the current PR. 
Since I have no more comments about this PR, I'll approve it as soon this question is decided.

**Review by Joao-L-S-Almeida — APPROVED** [review]

**Pull request merged**
